Common/include/geometry/primal_grid/CPrimalGrid.hpp

```diff
@@ -45,50 +46,60 @@
  */
 class CPrimalGrid {
 protected:
-  unsigned long *Nodes;         /*!< \brief Vector to store the global nodes of an element. */
+  vector<unsigned long> Nodes;         /*!< \brief Global node indices of the element. */
   unsigned long GlobalIndex;    /*!< \brief The global index of an element. */
```

Comment:
Boundary elements do not have a global index, but they have a DomainElement and vice versa.
So we can (if c++ allows) do:
```
union {
  unsigned long GlobalIndex;
  unsigned long DomainElement;
};
```